### ISSUE TITLE
[Jest Migration] polar/RadarSpec

### ISSUE THREAD

**Issue by proke03** [recharts/recharts#3236]
We want to move all tests to jest, RTL, and typescript. Issues will be made to encourage small chunks of refactoring done at once.

Refactor test/specs/polar/RadarSpec.js -> test-jest/polar/Radar.spec.ts.

https://github.com/recharts/recharts/blob/master/test/specs/polar/RadarSpec.js

Acceptance criteria:

All tests are in TS + RTL + Jest at Radar.spec.ts
RadarSpec.js is deleted
All tests pass

**Comment by akib22** [comment]
I am working on this issue. But I facing some problems. The `Radar` component does not accept `points` props as an array.
```ts
const data = [
  { x: 200, y: 230, cx: 250, cy: 250, angle: 30, radius: 60, value: 4 },
  { x: 300, y: 405, cx: 250, cy: 250, angle: 90, radius: 60, value: 4 },
  { x: 100, y: 600, cx: 250, cy: 250, angle: 150, radius: 60, value: 4 },
  { x: 90, y: 400, cx: 250, cy: 250, angle: 210, radius: 60, value: 4 },
];
render(
    <Surface width={500} height={500}>
      <Radar isAnimationActive={false} points={data} />
    </Surface>,
 );
```
In the above code snippet, I am getting this error.
![image](https://user-images.githubusercontent.com/48094649/214013034-3ccf353c-3d49-4aa7-ab16-a208eb1a6140.png)
I don't what to do, please help me.

**Comment by ckifer** [comment]
I'll fix this

**Comment by ckifer** [comment]
@akib22 you should be unblocked!

**Comment by akib22** [comment]
Hi @ckifer, I have a question. 
```ts
it('Render customized label when label is set to be a react element', () => {
  const CustomizedLabel = ({ x, y }) => (
    <text x={x} y={y} className="customized-label">
      test
    </text>
  );
  const { container } = render(
    <Surface width={500} height={500}>
      <Radar dataKey="y" isAnimationActive={false} points={data} label={<CustomizedLabel />} />
    </Surface>,
  );

  expect(container.querySelectorAll('.customized-label')).toHaveLength(data.length);
});
```
The `CustomizedLabel` component accepts `x and y` as props. When passing `CustomizedLabel` as a react element to `label` props, what should I pass to `CustomizedLabel` as props?  `CustomizedLabel` should get the props automatically when the `Radar` component renders with `points`.

**Comment by ckifer** [comment]
Not sure what you're asking. You are correct, the props should get automatically passed to your custom label component. You can probably console log them to see what all gets passed.

**Comment by akib22** [comment]
> You are correct, the props should get automatically passed to your custom label component.

Sorry for my poor explanation. I need to pass props when I call 'CustomizedLabel' as a react component (<CustomizedLabel />). Otherwise, I'll get type errors. What props should I pass? Internally 'CustomizedLabel' will get the props automatically so how can I externally pass the props?
I hope, now it is clear to you.

**Comment by ckifer** [comment]
You could try

```ts
label={(props) => <CustomizedLabel  {...props} />}
```

or

```ts
label={({ x, y }) => <CustomizedLabel x={x} y={y}  />}
```

**Comment by akib22** [comment]
Worked, Thanks @ckifer.

**Comment by ckifer** [comment]
Closed by #3261
